spyder/widgets/findreplace.py

```diff
@@ -444,10 +472,60 @@ def replace_find(self, focus_replace_text=False):
                                        QTextCursor.KeepAnchor)
                 else:
                     break
-                if not self.all_check.isChecked():
+                if not replace_all:
                     break
-            self.all_check.setCheckState(Qt.Unchecked)
             if cursor is not None:
                 cursor.endEditBlock()
             if focus_replace_text:
                 self.replace_text.setFocus()
+
+    @Slot()
+    def replace_find_all(self, focus_replace_text=False):
+        """Replace and find all matching occurrences"""
+        self.replace_find(focus_replace_text, replace_all=True)
+
+                
+    @Slot()
+    def replace_find_selection(self, focus_replace_text=False):
+        """Replace and find in the current selection"""
+        def _word_replacer(match):
+            """Used to replace only match objects that are whole
+            words when the user has selected whole word search"""
+            unchanged = match.group(0)
+            if match.start(0)>0 and re.search(r'\w', seltxt[match.start(0)-1], flags=word_flags):
+                return unchanged
+            if match.end(0)<len(seltxt) and re.search(r'\w', seltxt[match.end(0)+1], flags=word_flags):
+                return unchanged
+            if re.search(r'\W', unchanged, flags=word_flags):
+                return unchanged
+            return match.expand(replace_text) 
+
+        if (self.editor is not None):
+            replace_text = to_text_string(self.replace_text.currentText())
+            search_text = to_text_string(self.search_text.currentText())
+            pattern = search_text if self.re_button.isChecked() else None
+            case = self.case_button.isChecked()
+            words = self.words_button.isChecked()
+            re_flags = re.MULTILINE if case else re.IGNORECASE|re.MULTILINE
+
+            cursor = self.editor.textCursor()
+            cursor.beginEditBlock()
+            seltxt = to_text_string(self.editor.get_selected_text())
+            if not pattern:
+                pattern = re.escape(search_text)
+                replace_text = re.escape(replace_text)
+            if words:
+                #If whole words is checked we need to check that each match
+                #is actually a whole word before replacing
+                word_flags = 0 if case else re.IGNORECASE
+                replacement = re.sub(pattern, _word_replacer, seltxt, flags=re_flags)

```

Comment:
Why _word_replacer is needed? I think that this could be achieved changing the pattern, like other functions do:
```
if words:
    pattern = r"\b{}\b".format(patern)
```